### PR TITLE
first release

### LINKED ISSUE #1: Kuberentes Deployment to OAM converter
Migrated from https://github.com/oam-dev/rudr/issues/347

Ideally, we can covert a given kubernetes deployment object to a OAM defined application component and the required traits. Then the users can enjoy the benefit of separate concerns and supplying orthogonal operational capabilities without changing the app definition.

This would be supported in _migrate_ sub-command.

### PR DESCRIPTION
**Pull request by Diddaa**
first release of oamctl tool, this PR contains resource converter.

fixes #1

### CODE REVIEW

**Comment by wonderflow** [comment]
Awesome PR!

**Review by wonderflow — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,2 +1,138 @@
 # oamctl
-OAM command line tool
+
+oamctl is a tiny tool help oam user manage oam app.
+
+## Install
+
+``
```
Comment (hongchaodeng) [inline README.md]:
```suggestion
```
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,138 @@
...
+go build -o oamctl main.go
+``
```
Comment (hongchaodeng) [inline README.md]:
```suggestion
```
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,138 @@
 # oamctl
-OAM command line tool
+
+oamctl is a tiny tool help oam user manage oam app.
+
+## Install
+
+``
+go build -o oamctl main.go
```
Comment (hongchaodeng) [inline README.md]: use `go install`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,138 @@
 # oamctl
-OAM command line tool
+
+oamctl is a tiny tool help oam user manage oam app.
+
+## Install
+
```
Comment (hongchaodeng) [inline README.md:5]:
```suggestion
Run the following command to install _oamctl_:
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,138 @@
...
+``
+go build -o oamctl main.go
+``
+
+## Usage
```
Comment (hongchaodeng) [inline README.md]: This should not be put in README.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,138 @@
...
+      --config string      config file (default is $HOME/.oamctl.yaml)
+  -h, --help               help for oamctl
+  -n, --namespace string   operate namespace (default "default")
+  -t, --toggle             Help message for toggle
+
+Use "oamctl [command] --help" for more information about a command.
+
+```
+
+### oamctl migrate
```
Comment (hongchaodeng) [inline README.md]: Write a separate doc documenting each sub-command?

File: README.md [hunk 7]
```diff
@@ -1,2 +1,138 @@
...
+      --service string      name of service
+
+Global Flags:
+      --config string      config file (default is $HOME/.oamctl.yaml)
+  -n, --namespace string   operate namespace (default "default")
+
+```
+
+
+## example
```
Comment (hongchaodeng) [inline README.md]:
```suggestion
## Migrate to OAM from k8s deployment
```

File: examples/exist-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: legacy-deployment
```
Comment (hongchaodeng) [inline examples/exist-deployment.yaml]: maybe not use the name `legacy-*` but use `*-2migrate`

File: go.mod [hunk 9]
```diff
@@ -0,0 +1,19 @@
+module gitlab.alibaba-inc.com/cloudnativeapp/oamctl
```
Comment (hongchaodeng) [inline go.mod]: Change module name to `github.com/oam-dev/oamctl`

File: pkg/convert/root.go [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+
```
Comment (hongchaodeng) [inline pkg/convert/root.go:14]: remove space

**Comment by wonderflow** [comment]
@hongchaodeng  I think we could merge this initial PR and give some help.

**Comment by hongchaodeng** [comment]
@wonderflow 
If you want to take over this PR, here is the steps:
- Check out this change in local branch

```
git fetch origin pull/$ID/head:$BRANCHNAME
git checkout $BRANCHNAME
```

- Add your changes into a new commit
- Submit a new PR to replace this one.

**Comment by Diddaa** [comment]
@hongchaodeng I've fixed all the comments, thanks.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**